src/main/java/com/arpnetworking/tsdcore/sinks/KairosDbSink.java

```diff
@@ -73,17 +75,17 @@ public Object toLogValue() {
     }
 
     @Override
-    protected Collection<byte[]> serialize(final PeriodicData periodicData) {
+    protected Collection<SerializedDatum> serialize(final PeriodicData periodicData) {
         // Initialize serialization structures
-        final List<byte[]> completeChunks = Lists.newArrayList();
+        final List<SerializedDatum> completeChunks = Lists.newArrayList();
         final ByteBuffer currentChunk = ByteBuffer.allocate(_maxRequestSize);
+        final AtomicLong currentChunkPopulationSize = new AtomicLong(0L);
```

Comment:
Unless you need it to be safe for concurrent access, I suggest `LongAccumulator`; [Ref](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/LongAccumulator.html).

Reply:
Thanks. I couldn't think of the right construct to use. I changed it to a `LongAdder` which is a simplified extension of `LongAccumulator`

Reply:
👍